docs/tools/platformio.mdx

```diff
@@ -0,0 +1,128 @@
+---
+custom_edit_url: null
+description: The gate is a major tool of the Luos ecosystem.
+---
+
+import { customFields } from '/docusaurus.config.js';
+import Tooltip from '@site/src/components/Tooltip.js';
+import Image from '@site/src/components/Images.js';
+
+# PlatformIO
+
+## What is PlatformIO?
+
+PlatformIO is user-friendly and extensible integrated development environment with a set of professional development instruments, compiler and debugger. PlatformIO provide modern and powerful features to simplify the creation and delivery of embedded products
+That's the reason why Luos use PlatformIOs project structure for its examples, allowing user to use and share easily service and development on a unified environment. Also Luos engine and some tools like [Gate](/docs/tools/gate) or [Inspector](/docs/tools/inspector) are available on PlatformIOs registry! This registry allow user to link directly a project and a library, like Luos engine, by including dependencies in the configuration project file call 'Platformio.ini'
+learn more about <a href="https://platformio.org" target="_blank">PlatformIO and how to install PlatformIO<IconExternalLink width="10"></IconExternalLink></a>.
+
+## PlatformIO project structure
+
+Every PlatformIO project have a <a href="https://docs.platformio.org/en/latest/projectconf/section_platformio.html#directory-options" target="_blank">basic folder project structure<IconExternalLink width="10"></IconExternalLink></a> and must have at the root of the project a <a href="https://docs.platformio.org/en/latest/projectconf/section_platformio.html#generic-options" target="_blank">platformio.ini<IconExternalLink width="10"></IconExternalLink></a> file
+
+Checkout luos tutorial on <a href="https://www.luos.io/tutorials/pio" target="_blank">how create a Luos project with PlatformIo <IconExternalLink width="10"></IconExternalLink></a>.
+
+
+## Luos configuration for 'Platformio.ini'
+
+Some pre-compile definition may be use to configure Luos engine when its add in your project. those definition will be taking care of for Luos engine library using Luos script call at building a project with Luos engine.
+For more information, see the script source_filter_script.py in <a href="https://github.com/Luos-io/luos_engine" target="_blank" rel="external nofollow"> Luos engine <IconExternalLink width="10"></IconExternalLink> </a>.
+
+### Definition for Luos engine Library
+
+|      Definition       |                    Description                    |         Value         |
+| :-------------------: |-------------------------------------------------: | :-------------------: |
+|       LUOSHAL         | Define HAL family to use with Luos engine library |   see Luos engine HAL |
```

Comment:
Here in value you should explain what value are possible. Possible values are HAL folders name.

Reply:
Done for every value